packages/agoric-cli/src/commands/wallet.js

```diff
@@ -0,0 +1,113 @@
+// @ts-check
+/* eslint-disable func-names */
+/* global fetch, process */
+import {
+  iterateLatest,
+  makeCastingSpec,
+  makeFollower,
+  makeLeaderFromRpcAddresses,
+} from '@agoric/casting';
+import { Command } from 'commander';
+import { fmtRecordOfLines, simplePurseBalances } from '../lib/format.js';
+import { simpleOffers } from '../lib/psm.js';
+import { makeRpcUtils, networkConfig } from '../lib/rpc.js';
+import { getWalletState } from '../lib/wallet.js';
+
+// Ambient types. Needed only for dev but this does a runtime import.
+import '@agoric/vats/src/core/types.js';
+import '@agoric/vats/src/types.js';
+import { makeLeaderOptions } from '../lib/casting.js';
+import { normalizeAddress } from '../lib/keys.js';
+
+export const makeWalletCommand = async () => {
+  const wallet = new Command('wallet').description('wallet commands');
+
+  wallet
+    .command('send')
+    .description('send a prepared offer')
+    .requiredOption(
+      '--from [address]',
+      'address literal or name',
+      normalizeAddress,
+    )
+    .requiredOption('--offer [filename]', 'path to file with prepared offer')
+    .action(function () {
+      const { from, offer } = this.opts();
+      const { chainName, rpcAddrs } = networkConfig;
+
+      const cmd = `agd --node=${rpcAddrs[0]} --chain-id=${chainName} --from=${from} tx swingset wallet-action --allow-spend "$(cat ${offer})"`;
+
+      process.stdout.write('Run this interactive command in shell:\n\n');
+      process.stdout.write(cmd);
+      process.stdout.write('\n');
+    });
+
+  wallet
+    .command('list')
+    .description('list all wallets in vstorage')
+    .action(async function () {
+      const { vstorage } = await makeRpcUtils({ fetch });
+      const wallets = await vstorage.keys('published.wallet');
+      process.stdout.write(wallets.join('\n'));
+    });
+
+  wallet
+    .command('show')
+    .description('show current state')
+    .requiredOption(
+      '--from <address>',
+      'address literal or name',
+      normalizeAddress,
+    )
+    .action(async function () {
+      const opts = this.opts();
+      const { agoricNames, fromBoard, vstorage } = await makeRpcUtils({
+        fetch,
+      });
+      const state = await getWalletState(opts.from, fromBoard, {
+        vstorage,
+      });
+      const { assets, balances } = state;
+      // console.log(JSON.stringify(offers, null, 2));
+      // console.log(JSON.stringify({ offers, purses }, bigIntReplacer, 2));
+      const summary = {
+        balances: simplePurseBalances([...balances.values()], assets),
+        offers: simpleOffers(state, agoricNames),
+      };
+      process.stdout.write(fmtRecordOfLines(summary));
```

Comment:
`process.stdout` is ambient rather than explicit authority here.

Reply:
yeah, since this is package that doesn't export any utilities and only some bins, I think ambient authority is pragmatic. This package src is importing, whether it's in the entrypoint or lower doesn't make much difference I'm aware of.